src/functions.jl

```diff
@@ -42,8 +27,15 @@ abstract type AbstractVectorFunction <: AbstractFunction end
 The function that extracts the scalar variable referenced by `variable`, a `VariableIndex`.
 This function is naturally be used for single variable bounds or integrality constraints.
 """
-struct SingleVariable <: AbstractScalarFunction
-    variable::VariableIndex
+const SingleVariable = VariableIndex
+VariableIndex(vi::VariableIndex) = vi
+function Base.getproperty(vi::VariableIndex, s::Symbol)
+    if s == :value
```

Comment:
you should use `===`, see: https://discourse.julialang.org/t/overloading-getproperty-in-a-type-stable-manner/36638/2

Reply:
Thanks, I didn't know!